src/systemctl/systemctl.c

```diff
@@ -2433,12 +2433,25 @@ static int unit_file_find_path(LookupPaths *lp, const char *unit_name, char **un
 
         STRV_FOREACH(p, lp->search_path) {
                 _cleanup_free_ char *path;
+                _cleanup_free_ char *lpath = NULL;
+                int r;
 
                 path = path_join(arg_root, *p, unit_name);
                 if (!path)
                         return log_oom();
 
-                if (access(path, F_OK) == 0) {
+                r = chase_symlinks(path, arg_root, &lpath);
+                if (r == -ENOENT)
+                        continue;
+                else if (r == -ENOMEM)
+                        return log_oom();
+                else if (r < 0)
+                        return log_error_errno(r, "Failed to access path '%s': %m", path);
+                else if (r == 0) {
+                        *unit_path = lpath;
+                        lpath = NULL;
+                        return 1;
+                } else {
                         *unit_path = path;
                         path = NULL;
                         return 1;
```

Comment:
not that i matters much, but given that each if branch here ends in a "return" of some kind you can actually drop all the "else"s here, and just use plain "if" checks, if you follow what i mean.